store/sqlstore/store.go

```diff
@@ -318,6 +319,14 @@ func (ss *SqlStore) initConnection() {
 		}
 	}
 
+	// When WithInstance is used in migration, the underlying driver connections are not tracked.
+	// So we will have to open a fresh connection for migrations and explicitly close it when all is done.
+	var err error
+	dataSource, err = ss.appendMultipleStatementsFlag(dataSource)
+	if err != nil {
+		panic(err)
```

Comment:
We should not use `multiStatements` for the whole application, just during migrations. This means that we are forced to open a new DB instance for MySQL. This is expected.

I'd suggest to bring this back inside `migrate` and check if the driver is MySQL. If yes, then call this and `setupConnection`. Else just reuse `ss.GetMasterX().DB.DB`